### PR TITLE
Update `paddle.clamp`

### PR DESCRIPTION
**Pull request by willthefrog**
### PR types
Breaking changes 

### PR changes
APIs

### Describe

rename to `paddle.clip`
add fast path for dygraph mode
remove `out`
rename `input` -> `x`
update doc sample
also fix `cosine_similarity` conflict

### CODE REVIEW

File: python/paddle/tensor/math.py [hunk 1]
```diff
@@ -1302,53 +1302,50 @@ def addcmul(input, tensor1, tensor2, value=1.0, name=None):
...
 
     .. math::
 
         Out = MIN(MAX(x, min), max)
 
     Args:
-        input (Variable): An input N-D Tensor or LoDTensor
-            with data type float32, float64.
+	x (Tensor): An N-D Tensor with data type float32 or float64.
         min (float32|Variable): The lower bound with type ``float32`` or a ``Tensor``
```
Comment (jerrywgz) [inline python/paddle/tensor/math.py]: Variable -> Tensor here as well?

Comment (willthefrog) [reply python/paddle/tensor/math.py]: good catch,  done, thanks!

File: python/paddle/tensor/math.py [hunk 2]
```diff
@@ -1301,53 +1301,50 @@ def addcmul(input, tensor1, tensor2, value=1.0, name=None):
     return out
 
 
-def clamp(input, min=None, max=None, name=None):
+def clip(x, min=None, max=None, name=None):
     """
-	:alias_main: paddle.clamp
-	:alias: paddle.clamp,paddle.tensor.clamp,paddle.tensor.math.clamp
+        :alias_main: paddle.clip
+        :alias: paddle.clip,paddle.tensor.clip,paddle.tensor.math.clip
```
Comment (jzhang533) [inline python/paddle/tensor/math.py:1330]: 这两行写在docstring里的alias标签是可以去掉的。

Comment (willthefrog) [reply python/paddle/tensor/math.py:1330]: done

**Review by jzhang533 — DISMISSED** [review]
lgtm

**Review by kolinwei — DISMISSED** [review]
tensor.clamp倍tensor.clip替换，删除相关单测

**Review by kolinwei — DISMISSED** [review]

**Review by kolinwei — DISMISSED** [review]

**Review by kolinwei — DISMISSED** [review]

**Review by zhiqiu — DISMISSED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]
lgtm

**Review by kolinwei — APPROVED** [review]

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**